### PR TITLE
1214 upgrade to hocon 0.32.0

### PR DESCRIPTION
**Pull request by zmstone**
this is a follow up fix for https://github.com/emqx/emqx/pull/9437

verified with:

```
env EMQX_AUTHENTICATION__1='{mechanism="password_based",backend="mysql",server="localhost:3306",database="emqx",username="emqx",password="******",query="SELECT password_hash, salt, is_superuser FROM mqtt_user WHERE username = ${username} LIMIT 1",password_hash_algorithm={name=sha256,salt_position=prefix},enable=true}' \
    EMQX_AUTHENTICATION__1__server='"localhost:3333"' \
    ./_build/emqx/rel/emqx/bin/emqx consol
```
```
v5.0.11-g375e3de0(emqx@127.0.0.1)1> emqx:get_config([authentication]).
[#{backend => <<"mysql">>,database => <<"emqx">>,
   enable => true,mechanism => <<"password_based">>,
   password => <<"******">>,
   password_hash_algorithm =>
       #{name => <<"sha256">>,salt_position => <<"prefix">>},
   query =>
       <<"SELECT password_hash, salt, is_superuser FROM mqtt_user WHERE username = ${username} LIMIT 1">>,
   server => <<"localhost:3333">>,username => <<"emqx">>}]
```

### CODE REVIEW

File: apps/emqx_authn/src/emqx_authn.erl [hunk 1]
```diff
@@ -55,38 +58,39 @@ check_config(Config, Opts) ->
...
     end;
-do_check_config(Config, _Opts) when is_map(Config) ->
-    throw({invalid_config, "mechanism_field_required", Config});
-do_check_config(RawConf, Opts) ->
-    %% authentication conf is lazy type, when it comes from ENV, it is a string
-    %% EMQX_AUTHENTICATION__1="{mechanism=\"password_based\"...}"
-    case hocon:binary(RawConf, Opts) of
-        {ok, Conf} -> do_check_config(Conf, Opts);
-        {error, Reason} -> throw({invalid_config, Reason})
-    end.
```
Comment (thalesmg) [inline apps/emqx_authn/src/emqx_authn.erl:82]: Q: this clause can be removed because `remove_env_meta => true` has been added to `emqx_config:merge_envs`?  It's a bit to hard to follow hocon's magic :see_no_evil:

**Review by thalesmg — APPROVED** [review]

File: apps/emqx_authn/src/emqx_authn.erl (hunk 1 above)
Comment (zmstone) [reply apps/emqx_authn/src/emqx_authn.erl:82]: yes,
this clause was added wrong, it was not binary, but a tuple causing the function clause.
